zebrad/src/components/mempool/tests/vector.rs

```diff
@@ -956,6 +1083,8 @@ async fn setup(
         chain_tip_change.clone(),
     );
 
+    tokio::spawn(async move { while mempool_transaction_receiver.recv().await.is_ok() {} });
```

Comment:
Why do we need this?

Reply:
Dropping the receiver at the end of the `setup()` function could cause the mempool's `poll_ready()` method to return an error when trying to send transaction ids to a closed channel.